packages/office-ui-fabric-react/src/components/Layer/examples/Layer.Customized.Example.tsx

```diff
@@ -1,84 +1,52 @@
 
-export interface ILayerCustomizedExampleState {
-  showPanel: boolean;
-  trapPanel: boolean;
-}
+const layerHostClass = mergeStyles({
+  position: 'relative',
+  height: 400,
+  overflow: 'hidden',
+  border: '1px solid #ccc'
+});
 
-export class LayerCustomizedExample extends React.Component<{}, ILayerCustomizedExampleState> {
-  public state: ILayerCustomizedExampleState = {
-    showPanel: false,
-    trapPanel: false
-  };
-  // Use getId() to ensure that the ID is unique on the page.
-  // (It's also okay to use a plain string without getId() and manually ensure uniqueness.)
-  private _layerHostId: string = getId('layerHost');
+const focusTrapZoneProps: IFocusTrapZoneProps = {
+  isClickableOutsideFocusTrap: true,
+  forceFocusInsideTrap: false
+};
 
-  public render(): JSX.Element {
-    return (
-      <div>
-        <p>
-          A <code>Panel</code> is rendered, trapped in a specified container. Use <b>Show panel</b> to show/hide the panel (or click the X
-          to dismiss it). Use <b>Trap panel</b> to release the panel from its bounds.
-        </p>
-        <Toggle label="Show panel" inlineLabel checked={this.state.showPanel} onChange={this._onShowPanelChange} />
-        <Toggle label="Trap panel" inlineLabel checked={this.state.trapPanel} onChange={this._onTrapPanelChange} />
-        <Customizer
-          scopedSettings={
-            this.state.trapPanel
-              ? {
-                  Layer: {
-                    hostId: this._layerHostId
-                  }
-                }
-              : {}
-          }
-        >
-          {this.state.showPanel && (
-            <Panel
-              isOpen={true}
-              hasCloseButton={true}
-              headerText="Test"
-              focusTrapZoneProps={{
-                isClickableOutsideFocusTrap: true,
-                forceFocusInsideTrap: false
-              }}
-              onDismiss={this._onDismissPanel}
-            />
-          )}
-        </Customizer>
-        <LayerHost
-          id={this._layerHostId}
-          style={{
-            position: 'relative',
-            height: '400px',
-            overflow: 'hidden',
-            border: '1px solid #ccc'
-          }}
-        />
-      </div>
-    );
-  }
+export const LayerCustomizedExample: React.FunctionComponent = () => {
+  const [isPanelOpen, showPanel, dismissPanel] = useBoolean(false);
+  const [trapPanel, toggleTrapPanel] = useToggle(false);
 
-  private _onDismissPanel = (): void => {
-    this.setState({
-      showPanel: false
-    });
-  };
+  // Use useId() to ensure that the ID is unique on the page.
+  // (It's also okay to use a plain string and manually ensure uniqueness.)
+  const layerHostId = useId('layerHost');
 
-  private _onShowPanelChange = (event: React.FormEvent<HTMLElement | HTMLInputElement>, checked?: boolean): void => {
-    this.setState({
-      showPanel: !!checked
-    });
-  };
+  const scopedSettings = React.useMemo(() => {
```

Comment:
I almost want to have a rule that we extract hook logic to their own hooks and avoid writing hook logic in a render function. That way the render function stays as clean as possible.